Add Synthea postInstall file

josephsnyder opened the pull request:
Add another command line argument which will allow the user to install
a version of the Synthea Ingestor patch using a supplied DUZ value as
the installer.

Add information about the DUZ that needs to be supplied.  **At this point,
it could be added as a command line value for FOIA based instances, where the install will work as user 17. 
 But, it will  not on a VEHU instance due to user 17 not existing.**

josephsnyder committed:
Add Synthea postInstall file

Add another command line argument which will allow the user to install
a version of the Synthea Ingestor patch using a supplied DUZ value as
the installer.

Add information about the DUZ that needs to be supplied.  At this point,
it can be added as a command line value for FOIA based instances, but
not VEHU.

josephsnyder commented:
Okay!  We've got it down to one post install routine, installing  as user 1, which can be used like all the other postInstalls.  Please give this a try when you are ready.

ChristopherEdwards commented on Common/syntheaPostInstall.sh:
```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+mkdir /tmp/loader
+pushd /tmp/loader
+curl -fsSL --progress-bar https://github.com/OSEHRA/VistA-FHIR-Data-Loader/releases/download/0.1/VISTA_SYNTHETIC_DATA_LOADER_BUNDLE_0P2T1.KID.zip -o loader.zip
```
is there a way to use the github api to always get the latest version? may prevent future updates to this script.

shabiel replied:
No. We have to use specific KIDS builds in releases. We are not just getting the source code as there is some pre-install and post-install stuff that needs to be done.

ChristopherEdwards replied:
I'm referring to the latest release kids build not source code. Not a big issue, if we lock versions then we should probably have a comment here.

ChristopherEdwards commented on Common/syntheaPostInstall.sh:
```diff
@@ -0,0 +1,25 @@
+  system=2
+  instanceName=''
+fi
+if [ ! -d "VistA" ]; then
```
If i read this right we are in the /tmp/loader directory and looking for the VistA directory, I highly doubt that will ever exist in new installs, is there anywhere else that we download the VistA test repo and check for that instead?

ChristopherEdwards commented on Common/syntheaPostInstall.sh:
```diff
@@ -0,0 +1,25 @@
+pushd VistA/Scripts/
+# Installs as the DUZ=1 user
+python PatchSequenceApply.py -S $system -p /tmp/loader/Synthea $instanceName -l /tmp/ -i -n ALL -d 1
+popd
```
maybe we should popd before we pushd to VistA/Scripts?

shabiel replied:
Looks okay to me; but I will try it.

ChristopherEdwards replied:
i'm not saying what is written won't work, just seems messy.

shabiel commented:
Hmm... I don't feel comfortable doing that right now; plus I do not
anticipate making many new releases (maybe 1 or 2).

On Fri, Jan 4, 2019 at 11:48 AM Christopher Edwards
<notifications@github.com> wrote:
>
> @ChristopherEdwards commented on this pull request.
>
> ________________________________
>
> In Common/syntheaPostInstall.sh:
>
> > @@ -0,0 +1,25 @@
> +#!/bin/bash
> +mkdir /tmp/loader
> +pushd /tmp/loader
> +curl -fsSL --progress-bar https://github.com/OSEHRA/VistA-FHIR-Data-Loader/releases/download/0.1/VISTA_SYNTHETIC_DATA_LOADER_BUNDLE_0P2T1.KID.zip -o loader.zip
>
> I'm referring to the latest release kids build not source code. Not a big issue, if we lock versions then we should probably have a comment here.
>
> —
> You are receiving this because your review was requested.
> Reply to this email directly, view it on GitHub, or mute the thread.

josephsnyder committed:
Fix up directory and path usage

Ensure that only one pushd/popd is used to eliminate confusion.  Use the
vistaDir to call the PatchSequenceApply file within the /tmp/loader
directory.

shabiel committed:
Add M-Web-Server to Init file

shabiel reviewed (approved):
Everything now looks good
